src/herder/test/HerderTests.cpp

```diff
@@ -1867,7 +1880,7 @@ TEST_CASE("surge pricing with DEX separation", "[herder][txset]")
                        int64_t expectedDexBaseFee) {
         auto txSet = TxSetFrame::makeFromTransactions(txs, *app, 0, 0);
         size_t cntA = 0, cntB = 0, cntC = 0, cntD = 0;
-        auto resTxs = txSet->getTxsInApplyOrder();
+        auto resTxs = txSet->getResolvedFrame()->getTxsInApplyOrder();
```

Comment:
Should we also add a test where a catching up node receives future tx sets and resolves them correctly?

Reply:
Do we not have such a test? I was hoping this would be covered...

Reply:
well if he had such a test, we would have caught the issue that caused this refactor in the first place...

Reply:
Ah, sorry, I thought that you meant some generic test that deals with future tx sets - don't we have one? For Soroban issue specifically there is of course no test (and I'm honestly not sure how to set it up, given that it's kind of non-deterministic...)

Reply:
Ah, sorry, I thought that you meant some generic test that deals with future tx sets - don't we have one? For Soroban issue specifically there is of course no test (and I'm honestly not sure how to set it up, given that it's kind of non-deterministic...)

Reply:
Something like setting the bucketlist snapshot to every ledger, and starting a node that receives out of order future ledgers and tries to replay them should do the trick, right?

Reply:
I've added your catchup test, thanks for the help.